src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/UserServiceTest.kt

```diff
@@ -1,55 +1,109 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
 import org.mockito.Mockito.mock
 import org.mockito.kotlin.doReturn
 import org.mockito.kotlin.whenever
+import org.springframework.security.access.AccessDeniedException
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.ManageUsersClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.manageusers.model.UserDetailsDto.AuthSource
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isEqualTo
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isNotEqualTo
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.userDetails
-import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.userEmail
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.userEmailAddress
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.UserService.Companion.getClientAsUser
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.UserService.Companion.getServiceAsUser
 
 class UserServiceTest {
 
   private val manageUsersClient: ManageUsersClient = mock()
   private val userService = UserService(manageUsersClient)
 
   @Test
-  fun `getContactDetails should return ContactDetails for valid email username`() {
-    val username = "user@example.com"
+  fun `getServiceAsUser should return ContactDetails for valid email username`() {
+    val result = getServiceAsUser()
 
-    whenever(manageUsersClient.getUsersDetails(username)) doReturn userDetails(username, "Test User Name")
+    result.username isEqualTo "BOOK_A_VIDEO_LINK_SERVICE"
+    result.name isEqualTo "BOOK_A_VIDEO_LINK_SERVICE"
+    result.userType isEqualTo UserType.SERVICE
+    result.email isEqualTo null
+  }
 
-    val result = userService.getContactDetails(username)
+  @Test
+  fun `getClientAsUser should return ContactDetails for valid email username`() {
+    val result = getClientAsUser("client")
 
-    result!! isNotEqualTo null
-    result.name isEqualTo "Test User Name"
-    result.email isEqualTo username
+    result.username isEqualTo "client"
+    result.name isEqualTo "client"
+    result.userType isEqualTo UserType.SERVICE
+    result.email isEqualTo null
   }
 
   @Test
-  fun `getContactDetails should return ContactDetails for valid non-email username`() {
-    val username = "test_user"
+  fun `getUser should return user with PRISON type when authSource is nomis`() {
+    val username = "testUser"
+    whenever(manageUsersClient.getUsersDetails(username)) doReturn userDetails(username, "Test User", authSource = AuthSource.nomis)
+
+    val user = userService.getUser(username)
+
+    user isNotEqualTo null
```

Comment:
Not null checks not really needed.  You are asserting on values below them, it would NPE if null.